corehq/warehouse/models/facts.py

```diff
@@ -78,7 +79,38 @@ def dependencies(cls):
             FORM_STAGING_SLUG,
         ]
 
-# TODO: Write SyncLogFact
+
+class SyncLogFact(BaseFact, CustomSQLETLMixin):
+    '''
+    SyncLog Fact Table
+
+    Grain: sync_log_id
+    '''
+    slug = SYNCLOG_FACT_SLUG
+
+    sync_log_id = models.CharField(max_length=255)
+    sync_date = models.DateTimeField(null=True)
+
+    # these can be null per SyncLogStagingTable
+    domain = models.CharField(max_length=255, null=True)
+    user_id = models.CharField(max_length=255, null=True)
+
+    user_dim = models.ForeignKey(UserDim, on_delete=models.PROTECT)
+    domain_dim = models.ForeignKey(DomainDim, on_delete=models.PROTECT)
```

Comment:
why is it necessary to put the domain and user_id in the table as well as foreign keys to the user and domain dimensions?

Reply:
Yeah I had same question, Giovanni, Cal and I will [discuss this](This is what's being followed in the other Fact tables. ). This is what's being followed in the other Fact tables.

Reply:
I have the same question as simon, do you remember what resolution we came to? The only reason I could see for that is to allow some faster queries where all we need is the id, but that doesnt seem super userful, since rarely do we want the id and not the username.